pkg/microservice/aslan/core/common/service/environment.go

```diff
@@ -67,83 +69,154 @@ func FillProductTemplateValuesYamls(tmpl *templatemodels.Product, log *zap.Sugar
 		return 0, resp, ingressList, e.ErrListGroups.AddDesc(err.Error())
 	}
-
-	listServices, err := getter.ListServices(productInfo.Namespace, nil, kubeClient)
+	for _, v := range listDeployments {
+		workLoads = append(workLoads, &Workload{Name: v.Name, Spec: corev1.ServiceSpec{Selector: v.Spec.Selector.MatchLabels}, Type: setting.Deployment})
+	}
+	statefulSets, err := getter.ListStatefulSets(namespace, nil, kubeClient)
 	if err != nil {
-		log.Errorf("[%s][%s] create product record error: %v", envName, productName, err)
+		log.Errorf("[%s][%s] create product record error: %v", envName, namespace, err)
 		return 0, resp, ingressList, e.ErrListGroups.AddDesc(err.Error())
 	}
-
-	count := len(listServices)
+	for _, v := range statefulSets {
+		workLoads = append(workLoads, &Workload{Name: v.Name, Spec: corev1.ServiceSpec{Selector: v.Spec.Selector.MatchLabels}, Type: setting.StatefulSet})
+	}
+	// 对于workload过滤
+	if len(filter) > 0 {
+		workLoads = filter[0](workLoads)
+	}
+	count := len(workLoads)
 
 	// 分页
 	if page > 0 && perPage > 0 {
 		//将获取到的所有服务按照名称进行排序
-		sort.SliceStable(listServices, func(i, j int) bool { return listServices[i].Name < listServices[j].Name })
+		sort.SliceStable(workLoads, func(i, j int) bool { return workLoads[i].Name < workLoads[j].Name })
 
 		start := (page - 1) * perPage
 		if start >= count {
-			listServices = nil
+			workLoads = nil
 		} else if start+perPage >= count {
-			listServices = listServices[start:]
+			workLoads = workLoads[start:]
 		} else {
-			listServices = listServices[start : start+perPage]
+			workLoads = workLoads[start : start+perPage]
 		}
 	}
 
-	for _, service := range listServices {
+	for _, workload := range workLoads {
 		wg.Add(1)
 
-		go func(service *corev1.Service) {
+		go func(workload *Workload) {
 			defer func() {
 				wg.Done()
 			}()
 
+			tmpProductName := workload.ProductName
+			if tmpProductName == "" && productName != "" {
+				tmpProductName = productName
+			}
+
 			productRespInfo := &ServiceResp{
-				ServiceName: service.Name,
-				EnvName:     envName,
-				ProductName: productName,
-				Type:        setting.K8SDeployType,
+				ServiceName:  workload.Name,
+				EnvName:      workload.EnvName,
+				Type:         setting.K8SDeployType,
+				WorkLoadType: workload.Type,
+				ProductName:  tmpProductName,
 			}
 
-			selector := labels.SelectorFromValidatedSet(service.Spec.Selector)
-			productRespInfo.Status, productRespInfo.Ready, productRespInfo.Images = queryExternalPodsStatus(productInfo.Namespace, selector, kubeClient, log)
+			selector := labels.SelectorFromValidatedSet(workload.Spec.Selector)
+			productRespInfo.Status, productRespInfo.Ready, productRespInfo.Images = queryExternalPodsStatus(namespace, selector, kubeClient, log)
 
-			if ingresses, err := getter.ListIngresses(productInfo.Namespace, selector, kubeClient); err == nil {
+			if ingresses, err := getter.ListIngresses(namespace, selector, kubeClient); err == nil {
```

Comment:
we can not get workload related ingress records in this way, just remove it for now